cpp/src/slice2cpp/Gen.cpp

```diff
@@ -3117,15 +3166,33 @@ Slice::Gen::InterfaceVisitor::visitOperation(const OperationPtr& p)
     if (comment)
     {
         OpDocParamType pt = (amd || p->hasMarshaledResult()) ? OpDocInParams : OpDocAllParams;
-        StringList postParams, returns;
+        StringList postParams;
+        StringList returns;
         if (amd)
         {
-            postParams.push_back("@param " + responsecbParam + " The response callback.");
+            if (p->hasMarshaledResult())
+            {
+                postParams.push_back(
+                    "@param " + responsecbParam + " The response callback. It accepts a marshaled result.");
+            }
+            else
+            {
+                if (ret || !outParams.empty())
```

Comment:
We have a more readable function for this check.
```suggestion
                if (p->returnsAnyValues())
```